File: scm-java/src/main/scheme/primitives/PrimitiveGetOutputBytevector.java
package scheme.primitives;

import scheme.*;

public class PrimitiveGetOutputBytevector extends Primitive {
    @Override public String name() { return "get-output-bytevector"; }
    @Override public String info() {
        return "Syntax: (get-output-bytevector port)\n" +
               "Library: (scheme base)\n" +
               "Description: Returns a bytevector consisting of the bytes that have been output to the given bytevector output port (created with open-output-bytevector).\n" +
               "Example:\n" +
               "  (let ((p (open-output-bytevector)))\n" +
               "    (write-u8 65 p)\n" +
               "    (get-output-bytevector p)) => #u8(65)";
    }

    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 1, 1);
        return Value.asBinaryOutputPort(arguments[0]).getBytes();
    }
}

File: scm-csharp/scheme/primitives/PrimitiveGetOutputBytevector.cs
namespace scheme;

public class PrimitiveGetOutputBytevector : Primitive
{
    public override string Name() => "get-output-bytevector";
    public override string Info() =>
        "Syntax: (get-output-bytevector port)\n" +
        "Library: (scheme base)\n" +
        "Description: Returns a bytevector consisting of the bytes that have been output to the given bytevector output port (created with open-output-bytevector).\n" +
        "Example:\n" +
        "  (let ((p (open-output-bytevector)))\n" +
        "    (write-u8 65 p)\n" +
        "    (get-output-bytevector p)) => #u8(65)";

    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 1, 1);
        return Value.AsBinaryOutputPort(arguments[0]).GetBytes();
    }
}
